Spec/RealtimeClientConnection.swift

```diff
@@ -3468,6 +3468,88 @@ class RealtimeClientConnection: QuickSpec {
+                        done()
+                    }
+                }
+
+                client.connection.once(.Connected) { stateChange in
+                    fail("Should not emit a Connected state")
+                }
+
+                waitUntil(timeout: testTimeout) { done in
+                    client.connection.once(.Update) { stateChange in
+                        guard let stateChange = stateChange else {
+                            fail("ConnectionStateChange is nil"); done(); return
+                        }
+                        expect(stateChange.reason).to(beNil())
+                        expect(client.connection.state).to(equal(ARTRealtimeConnectionState.Connected))
+                        expect(stateChange.current).to(equal(ARTRealtimeConnectionState.Connected))
+                        expect(stateChange.current).to(equal(stateChange.previous))
+                        done()
+                    }
+
+                    client.auth.authorize(nil, options: nil) { tokenDetails, error in
+                        expect(tokenDetails).toNot(beNil())
+                        expect(error).to(beNil())
+                    }
+                }
+            }
+
+            // RTN24
+            it("should set the Reason attribute based on the Error member of the CONNECTED ProtocolMessage") {
+                let options = AblyTests.commonAppSetup()
+                options.useTokenAuth = false
+                let client = ARTRealtime(options: options)
+                defer { client.dispose(); client.close() }
+
+                waitUntil(timeout: testTimeout) { done in
+                    client.connection.once(.Connected) { stateChange in
+                        expect(stateChange?.reason).to(beNil())
+                        done()
+                    }
+                }
+
+                client.connection.once(.Connected) { stateChange in
+                    fail("Should not emit a Connected state")
+                }
+
+                waitUntil(timeout: testTimeout) { done in
+                    client.connection.once(.Update) { stateChange in
+                        guard let stateChange = stateChange else {
+                            fail("ConnectionStateChange is nil"); done(); return
+                        }
+                        guard let error = stateChange.reason else {
+                            fail("Reason error is nil"); done(); return
+                        }
+                        expect(error.code) == 40102
+                        expect(client.connection.state).to(equal(ARTRealtimeConnectionState.Connected))
+                        expect(stateChange.current).to(equal(ARTRealtimeConnectionState.Connected))
+                        expect(stateChange.current).to(equal(stateChange.previous))
+                        done()
+                    }
+
+                    client.auth.authorize(nil, options: nil) { tokenDetails, error in
```

Comment:
This will not result in a new CONNECTED message unfortunately.  The authorize may fail, but the connection does not change so no new CONNECTED message is sent.  Instead, in this instance you may need to fabricate a CONNECTED message I am afraid with an embedded error and you will simply need to check that the error reason is set from that.  I know we try not to fabricate, but in this instance I cannot think of a way around that.